system/jlib/jlz4.cpp

```diff
@@ -0,0 +1,316 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "platform.h"
+#include "jfcmp.hpp"
+#include "jlz4.hpp"
+#include "lz4.h"
+
+/* Format:
+    size32_t totalexpsize;
+    { size32_t subcmpsize; bytes subcmpdata; }
+    size32_t trailsize; bytes traildata;    // unexpanded
+*/
+
+class jlib_decl CLZ4Compressor : public CFcmpCompressor
+{
+    inline void setinmax()
+    {
+        inmax = blksz-outlen-sizeof(size32_t);
+        if (inmax<256)
+            trailing = true;    // too small to bother compressing
+        else
+        {
+            trailing = false;
+            size32_t slack = LZ4_COMPRESSBOUND(inmax) - inmax;
+            int inmax2 = inmax - (slack + sizeof(size32_t));
+            if (inmax2<256)
+                trailing = true;
+            else
+                inmax = inmax2;
+        }
+    }
+
+    inline void flushcommitted()
```

Comment:
minor: as base is virtual, clearer if this was declared virtual too, but think inline should be removed (see comment on base method)